### PR TITLE
fix(components):[numeric]Fixed a bug where manually entering values does not change according to step after fixing the stepStrictly property setting

### PR DESCRIPTION
**Pull request by fanbingbing16**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Manually entering values after setting the stepStrict property will not change according to the steps

Issue Number: #964 

## What is the new behavior?
After setting the stepStrict property, manually input the value and automatically change it according to the steps

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by Huangyilin19** [comment]
@fanbingbing16  检视意见麻烦尽快处理下，有什么不了解的或自己有别的想法也可以留个言。

**Comment by fanbingbing16** [comment]
> @fanbingbing16 检视意见麻烦尽快处理下，有什么不了解的或自己有别的想法也可以留个言。

请问检视意见在哪里？我没有找到

File: packages/renderless/src/numeric/index.ts [hunk 1]
```diff
@@ -336,11 +336,23 @@ export const handleInput =
...
-    const value = event.target.value
+    const value = event.target?.value === '-' ? 0 : event.target?.value
+    if (props.stepStrictly) {
+        const previousValue = +((props.mouseWheel ? state.displayValue : props.modelValue) || 0)
+        if (Math.abs(previousValue - value) % +props.step === 0) return api.setCurrentValue(value)
 
-    api.setCurrentValue(value === '-' ? 0 : value)
+        if (previousValue < 0 || value < 0)
+            return api.setCurrentValue(
+                -Math.round(Math.abs(previousValue - value) / +props.step) * +props.step + previousValue
```
Comment (gimmyhehe) [inline packages/renderless/src/numeric/index.ts]: 根据开发规范，不推荐使用+props.step的方式转化为数字类型，此处推荐使用Number(props.step)转化

File: packages/renderless/src/numeric/index.ts [hunk 2]
```diff
@@ -336,11 +336,23 @@ export const handleInput =
...
+            )
+
+        return api.setCurrentValue(
+            Math.round(Math.abs(previousValue - value) / +props.step) * +props.step + previousValue
```
Comment (gimmyhehe) [inline packages/renderless/src/numeric/index.ts]: 此处同上

Comment (fanbingbing16) [reply packages/renderless/src/numeric/index.ts]: 已改好

File: packages/renderless/src/numeric/index.ts [hunk 3]
```diff
@@ -336,11 +336,23 @@ export const handleInput =
...
+  ({ api, state, props }: Pick<INumericRenderlessParams, 'api' | 'state' | 'props'>) =>
   (event: Event): void => {
-    const value = event.target.value
+    const value = event.target?.value === '-' ? 0 : event.target?.value
+    if (props.stepStrictly) {
+        const previousValue = +((props.mouseWheel ? state.displayValue : props.modelValue) || 0)
+        if (Math.abs(previousValue - value) % +props.step === 0) return api.setCurrentValue(value)
 
-    api.setCurrentValue(value === '-' ? 0 : value)
+        if (previousValue < 0 || value < 0)
```
Comment (Huangyilin19) [inline packages/renderless/src/numeric/index.ts]: 346-353 这一块代码逻辑有问题，应该根据**变化的差值**而不是原值或现值是否为负。可以参考以下思路：
```
const step = Number(props.step)
const difference = previousValue - value
const sign = difference >= 0 ? 1: -1
return api.setValue(sign * Math.round(Math.abs(difference) / step) *  step + previousValue )

```

File: packages/renderless/src/numeric/index.ts [hunk 4]
```diff
@@ -336,11 +336,23 @@ export const handleInput =
   }
 
 export const handleInputChange =
-  ({ api }: Pick<INumericRenderlessParams, 'api'>) =>
+  ({ api, state, props }: Pick<INumericRenderlessParams, 'api' | 'state' | 'props'>) =>
   (event: Event): void => {
-    const value = event.target.value
+    const value = event.target?.value === '-' ? 0 : event.target?.value
+    if (props.stepStrictly) {
+        const previousValue = +((props.mouseWheel ? state.displayValue : props.modelValue) || 0)
```
Comment (Huangyilin19) [inline packages/renderless/src/numeric/index.ts]: 推荐使用Number()方法转化字符串， +操作符比较费解，代码易读性差些。
下同。

Comment (fanbingbing16) [reply packages/renderless/src/numeric/index.ts]: 已修改

**Review by gimmyhehe — APPROVED** [review]

**Review by Huangyilin19 — APPROVED** [review]

**Review by zzcr — APPROVED** [review]

**Pull request merged**